dexlib2/src/main/java/com/android/tools/smali/dexlib2/dexbacked/DexBackedDexFile.java

```diff
@@ -204,7 +204,9 @@ public static DexBackedDexFile fromInputStream(@Nullable Opcodes opcodes, @Nonnu
             throws IOException {
         DexUtil.verifyDexHeader(is);
 
-        byte[] buf = ByteStreams.toByteArray(is);
+        int fileSize = is.available();
+        byte[] buf = new byte[fileSize];
+        is.read(buf);
```

Comment:
This code pattern is here twice. Maybe add a FileUtils/ByteStreamUtils to util.

Reply:
InputStream.available does not read the file in full, so this is not equivalent to the full read done by ByteStreams.toByteArray.

Reply:
I think InputStream.readAllBytes is a valid replacement.

Reply:
Included toByteArray, readFully and skipFully to InputStreamUtil